### PR TITLE
Ms.parallel pool t

### PR DESCRIPTION
**Pull request by mariano54**
This used to be the slowest test.

### CODE REVIEW

File: tests/pools/test_pool_rpc.py [hunk 1]
```diff
@@ -400,7 +398,7 @@ async def test_create_multiple_pool_wallets(self, one_wallet_node_and_rpc, fee,
 
         # Test creation of many pool wallets. Use untrusted since that is the more complicated protocol, but don't
         # run this code more than once, since it's slow.
-        if fee == 0 and not trusted:
+        if not trusted:
```
Comment (arvidn) [inline tests/pools/test_pool_rpc.py:401]: this change seems unrelated

File: tests/pools/test_pool_rpc.py [hunk 2]
```diff
@@ -165,9 +163,9 @@ async def setup(two_wallet_nodes, bt, self_hostname):
 
 class TestPoolWalletRpc:
     @pytest.mark.asyncio
-    @pytest.mark.parametrize("trusted", [True, False])
-    @pytest.mark.parametrize("fee", [0, FEE_AMOUNT])
-    async def test_create_new_pool_wallet_self_farm(self, one_wallet_node_and_rpc, fee, trusted, self_hostname):
+    @pytest.mark.parametrize("trusted_and_fee", [(True, FEE_AMOUNT), (False, 0)])
```
Comment (arvidn) [inline tests/pools/test_pool_rpc.py:166]: how can we know that these are unrelated? and did you pick this combination arbitrarily?

File: tests/pools/test_pool_rpc.py [hunk 3]
```diff
@@ -693,10 +691,14 @@ async def test_absorb_pooling(self, one_wallet_node_and_rpc, fee, trusted, bt, s
             assert (250000000000 + fee) in [tx.additions[0].amount for tx in tx1]
 
     @pytest.mark.asyncio
-    @pytest.mark.parametrize("trusted", [True])
-    @pytest.mark.parametrize("fee", [0])
-    async def test_self_pooling_to_pooling(self, setup, fee, trusted, self_hostname):
-        """This tests self-pooling -> pooling"""
+    @pytest.mark.parametrize("trusted_and_fee", [(True, 0), (False, 0)])
```
Comment (arvidn) [inline tests/pools/test_pool_rpc.py:694]: `fee` is always 0 here...

File: tests/pools/test_pool_rpc.py [hunk 4]
```diff
@@ -793,8 +795,8 @@ async def tx_is_in_mempool(wid, tx: TransactionRecord):
                 fetched: Optional[TransactionRecord] = await client.get_transaction(wid, tx.name)
                 return fetched is not None and fetched.is_in_mempool()
 
-            await time_out_assert(5, tx_is_in_mempool, True, wallet_id, join_pool_tx)
-            await time_out_assert(5, tx_is_in_mempool, True, wallet_id_2, join_pool_tx_2)
+            await time_out_assert(10, tx_is_in_mempool, True, wallet_id, join_pool_tx)
+            await time_out_assert(10, tx_is_in_mempool, True, wallet_id_2, join_pool_tx_2)
```
Comment (arvidn) [inline tests/pools/test_pool_rpc.py:799]: was this flaky? I would expect speeding up the tests would mean shorter timeouts

**Review by arvidn — COMMENTED** [review]
the tests that take this long, do they create plots?
https://github.com/Chia-Network/chia-blockchain/runs/5763094009?check_suite_focus=true#step:9:82

File: tests/pools/test_pool_rpc.py (hunk 4 above)
Comment (mariano54) [reply tests/pools/test_pool_rpc.py:799]: I don't think they were flaky, but when running with -n 4 I saw this fail, so i increased it. I think creating plots was stalling the cores and therefore there were timeouts in the node.

**Comment by mariano54** [comment]
> the tests that take this long, do they create plots? https://github.com/Chia-Network/chia-blockchain/runs/5763094009?check_suite_focus=true#step:9:82

Yes but that's not the reason for the slowdown. That test in particular creates many blocks and wallets so it's very slow.

File: tests/pools/test_pool_rpc.py (hunk 3 above)
Comment (arvidn) [reply tests/pools/test_pool_rpc.py:694]: I see. in fact, in the previous version we only test this with `trusted=True` and `fee=0`. You added one more case.
But `fee` should probably not be a parameter at all to this test then

File: tests/pools/test_pool_rpc.py (hunk 1 above)
Comment (arvidn) [reply tests/pools/test_pool_rpc.py:401]: I see. so, technically, this is a separate, independent, test parameter.

**Review by arvidn — APPROVED** [review]

File: tests/pools/test_pool_rpc.py (hunk 1 above)
Comment (mariano54) [reply tests/pools/test_pool_rpc.py:401]: yeah good point, I guess i can parametrize that differently

File: tests/build-workflows.py [hunk 5]
```diff
@@ -83,7 +83,10 @@ def generate_replacements(conf, dir):
         ] = "# Omitted checking out blocks and plots repo Chia-Network/test-cache"
     if not conf["install_timelord"]:
         replacements["INSTALL_TIMELORD"] = "# Omitted installing Timelord"
-    replacements["PYTEST_PARALLEL_ARGS"] = " -n 4" if conf["parallel"] else " -n 0"
+    if conf.get("custom_parallel_n", None):
```
Comment (altendky) [inline tests/build-workflows.py:86]: Instead of having it be conditionally present we can set defaults in `testconfig.py`.

https://github.com/Chia-Network/chia-blockchain/blob/8833cc351c49a7e9340626c7c9e4715a0ecd3626/tests/testconfig.py#L1-L12

Comment (mariano54) [reply tests/build-workflows.py:86]: I didn't see that file, thank you
